internal/cmd/integration-tests/tests/beyla/beyla_test.go

```diff
@@ -2,22 +2,28 @@ package main
 
 import (
 	"testing"
+
+	"github.com/grafana/alloy/internal/cmd/integration-tests/common"
 )
 
 func TestBeylaMetrics(t *testing.T) {
-	// TODO this test is broken and needs deeper investigation
-	//var beylaMetrics = []string{
-	//	"beyla_internal_build_info",                // check that internal Beyla metrics are reported
-	//	"http_server_request_duration_seconds_sum", // check that the target metrics are reported
-	//}
-	//common.MimirMetricsTest(t, beylaMetrics, []string{}, "beyla")
+	var beylaMetrics = []string{
+		"beyla_internal_build_info",                // check that internal Beyla metrics are reported
+		"http_server_request_duration_seconds_sum", // check that the target metrics are reported
+	}
+	common.MimirMetricsTest(t, beylaMetrics, []string{}, "beyla")
 }
 
 func TestBeylaTraces(t *testing.T) {
-	// TODO this test is broken and needs deeper investigation
 	// Test that traces are being generated and sent to Tempo
-	//tags := map[string]string{
-	//	"service.name": "main", // This should match the instrumented app
-	//}
-	// common.TracesTest(t, tags, "beyla")
+	tags := map[string]string{
+		"service.name": "main", // This should match the instrumented app
+	}
+	common.TracesTest(t, tags, "beyla")
+}
+
+func TestBeylaMetricsAndTraces(t *testing.T) {
+	// Run both metrics and traces tests together
+	t.Run("metrics", TestBeylaMetrics)
+	t.Run("traces", TestBeylaTraces)
```

Comment:
```suggestion
```

Since the tests share the same backend testing them individually is the same as testing them together. Having both makes test failures take longer to uncover.